may dev update (#124)
irods/irods.org: content/posts/irods-development-update-may-2026.md

Title: iRODS Development Update: May 2026
Date: 2026-05-19 11:00
Author: Kory Draughn
Slug: irods-development-update-may-2026
Status: published

Hello reader,

The list of accepted talks for this year's User Group Meeting are now public. You can view the list and register for the event at <https://irods.org/ugm2026>.

Work on Logical Quotas is complete and has been merged into the server. It is the second plugin to be absorbed into the server and aligns with the goals of the 2025 roadmap.

Work to support FIPS-enabled environments is making steady progress. We've added the ability to configure SSL/TLS via the `setup_irods.py` script. The setup script can also generate self-signed certificates.

Using only the setup script, we were able to configure a server to use only the new **irods** authentication scheme. This feature needs a bit more time to cook, so it will not be part of the 5.1.0 release.

> Work to make the resource-rebalance operation continue where there are no good replicas of a data object is complete and has been merged. This enhancement should make management a little easier for administrators moving forward.

Reviewer: ```suggestion
Work to make the resource-rebalance operation continue where there are no good replicas of a data object is complete and has been merged. This enhancement should make management easier for large deployments.
```

Author: Done.

The univmss resource plugin now supports escaping single quotes in physical paths. This feature is disabled by default. Administrators can enable it through the context string.

To strengthen security, the server has been updated to disallow running with root privileges. It reports UID/GID information to help administrators in debugging situations. The server also reports when it is launched by a user who does not own the service account home directory.

> The iCommands now perform a compatibility check against the server. If the major version numbers do not match, a warning is printed to stderr. For situations where the warning negatively impacts scripts and automation, we've provided a way for users to disable the warning for all iCommands.

Reviewer: ```suggestion
The iCommands now perform a compatibility check against the server. If the major version numbers do not match, a warning is printed to stderr. For situations where the warning negatively impacts scripts and automation, we've provided a way for users to suppress the warning for all iCommands.
```

Author: Done.

The AMQP Audit rule engine plugin is undergoing a significant rework. Several configuration options are being exposed and connection management is being overhauled.

> A bug in the S3 resource plugin's ability to handle parallel transfer over port 1247 was discovered through the use of the HTTP API. That has bug has been fixed and will be included in the next release of the plugin.

Reviewer: ```suggestion
A bug in the S3 resource plugin's ability to handle parallel transfer over port 1247 was discovered through the use of the HTTP API. The bug has been fixed and will be included in the next release of the plugin.
```

Author: Done.
